assets/charts/components/azure-arc-onboarding/templates/register.yaml

```
@@ -0,0 +1,30 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+    "helm.sh/hook": "post-install"
+  name: azure-arc-register
+spec:
+  template:
+    spec:
+      serviceAccountName: azure-arc-onboarding
+      containers:
+      - image: quay.io/kinvolk/az-cli-with-helm:v0.1
+        name: install-azure-arc
+        args:
+          - register
+        env:
+          - name: AZURE_APPLICATION_CLIENT_ID
+            value: {{ required "applicationClientID is required" .Values.applicationClientID }}
```

I'd consider pipe notation, as I think it's more readable, as you first care about the value which will be placed (subject), then on the eventual condition that it must not be empty (rule).

Also applies to other uses of `required` function.
```suggestion
            value: {{ .Values.applicationClientID | required "applicationClientID is required"  }}
```